crates/sparrow-runtime/src/execute/operation/spread.rs

```diff
@@ -1708,13 +1896,84 @@ impl SpreadImpl for UnlatchedUInt64ListSpread {
     }
 }
 
+#[derive(serde::Serialize, serde::Deserialize, Debug)]
+struct UnlatchedMapSpread;
+
+impl ToSerializedSpread for UnlatchedMapSpread {
+    fn to_serialized_spread(&self) -> SerializedSpread<'_> {
+        SerializedSpread::UnlatchedMap(Boo::Borrowed(self))
+    }
+}
+
+impl SpreadImpl for UnlatchedMapSpread {
```

Comment:
There is probably a way to implement `LatchedMapSpread` using `concat_take` similar to what we did with first/last.